### ISSUE TITLE
No watch statistics

### ISSUE THREAD

**Issue by BrendenCA** [Tautulli/Tautulli#1149]
Version: 1.4.25  

Branch:master

Commit hash:cfd30996264b7e9fe4ef87f02d1cc52d1ae8bfca

Operating system: Debian 9.1

Python version: 2.7.13 (default, Jan 19 2017, 14:48:08) [GCC 6.3.0 20170118]

What you did?
Opened the plexpy home page

What happened?
There is no watch statistics last 30 days.

I believe I ran plexpy as another user for a period of time, by mistake. I chown the directories /var/lib/plexpy and /opt/plexpy

Everytime I open up the home page this pops up in the log.

2017-11-20 10:33:58 - WARNING :: CP Server Thread-8 : PlexPy DataFactory :: Unable to execute database query for get_home_stats: most_concurrent: 'int' object has no attribute '__getitem__'.

Can you help me? Don't want to lose all the statistics i've collected.
Also, the rest of the statistics show up fine Users/History/Graphs tab, issue is only with the watch statistics.

Thanks,

**Comment by ghost** [comment]
Sirwan Palani


Den 20. nov. 2017 kl. 06.15 skrev Brenden <notifications@github.com<mailto:notifications@github.com>>:


Version: 1.4.25

Branch:master

Commit hash:cfd30996264b7e9fe4ef87f02d1cc52d1ae8bfca

Operating system: Ubuntu 15.04

Python version: 2.7.13 (default, Jan 19 2017, 14:48:08) [GCC 6.3.0 20170118]

What you did?
Opened the plexpy home page

What happened?
There is no watch statistics last 30 days.

I believe I ran plexpy as another user for a period of time, by mistake. I chown the directories /var/lib/plexpy and /opt/plexpy

Everytime I open up the home page this pops up in the log.

2017-11-20 10:33:58 - WARNING :: CP Server Thread-8 : PlexPy DataFactory :: Unable to execute database query for get_home_stats: most_concurrent: 'int' object has no attribute 'getitem'.

Can you help me? Don't want to lose all the statistics i've collected.
Also, the rest of the statistics show up fine Users/History/Graphs tab, issue is only with the watch statistics.

Thanks,

**Comment by BrendenCA** [comment]
@JonnyWong16 
Going by the error I decided to check the types and noticed this 
https://github.com/JonnyWong16/plexpy/blob/cfd30996264b7e9fe4ef87f02d1cc52d1ae8bfca/plexpy/datafactory.py#L711

last_start is of int type but last_start[:-1] is being used later on

I changed it to last_start = "0"

Seems to have worked.

**Comment by BrendenCA** [comment]
@JonnyWong16
I figured out the issue,
Turns out my first entry for direct stream "copy" started and stopped at the same time, thus when sorting the stopped entry came infront of the started, resulting in last_start not being initialized to a str value.

https://github.com/JonnyWong16/plexpy/blob/cfd30996264b7e9fe4ef87f02d1cc52d1ae8bfca/plexpy/datafactory.py#L705
I think adding 'A' to started and 'B' to stopped is what's needed. Opposite of what it is currently.
